### ISSUE TITLE
Mbt build command getting stuck on a windows machine

### ISSUE THREAD

**Issue by fbunsmann** [SAP/cloud-mta-build-tool#800]
We currently have the issue that our `mbt` build is sometimes getting stuck on a windows machine. This seems to only happen while packaging a module build result into the temporary `mta_build_tmp` folder. For me, it gets stuck e. g. while building a module like the following. Unfortunately, the error is not reproducible in a stable manner and only occurs from time to time.

**Module Definition**:
```
modules:
- name: rm-db
  type: nodejs
  path: db
  ...
```

with the following `package.json`

```json
{
	"name": "rm-db-deployer",
	"dependencies": {
		"@sap/hdi-dynamic-deploy": "1.7.2"
	},
	"engines": {
		"node": "^10.0.0"
	},
	"scripts": {
		"postinstall": "node collectArtifacts",
		"start": "node node_modules/@sap/hdi-dynamic-deploy/"
	}
}
```

Here is the relevant part from the build log. I hade to remove some parts. Notice that there are no dots behind the packaging command for the `rm-db` module. This is why we think the build is getting stuck at that point. I manually aborted the build after a few minutes with no additional log output. Windows Task Manager did not show any CPU utilization related to the `mbt build` at that point.

```
> mbt build --mtar <HIDDEN>

[2020-11-13 14:04:19]  INFO Cloud MTA Build Tool version 1.0.16
[2020-11-13 14:04:19]  INFO generating the "Makefile_20201113140419.mta" file...
[2020-11-13 14:04:19]  INFO done
[2020-11-13 14:04:19]  INFO executing the "make -f Makefile_20201113140419.mta p=cf mtar=<HIDDEN> strict=true mode=" command...
[2020-11-13 14:04:19] DEBUG the executable file is at "<HIDDEN>"
[2020-11-13 14:04:20]  INFO validating the MTA project
[2020-11-13 14:04:20]  INFO running the "before-all" build...
[2020-11-13 14:04:20]  INFO executing the "npm install --production" command...
[2020-11-13 14:04:20] DEBUG the executable file is at "C:\Program Files\nodejs\npm.cmd"

<<< a (large) cds build is executed >>>

[cds] - build completed in: 30018.092ms

[2020-11-13 14:05:02]  INFO validating the MTA project
[2020-11-13 14:05:02]  INFO building the "rm-db" module...
[2020-11-13 14:05:02]  INFO executing the "npm install --production" command...
[2020-11-13 14:05:02] DEBUG the executable file is at "C:\Program Files\nodejs\npm.cmd"

<<< some other native artifacts are copied over to the db folder during npm postinstall >>> 

.[2020-11-13 14:05:07]  INFO the build results of the "rm-db" module will be packaged and saved in the "<HIDDEN>_mta_build_tmp\rm-db" folder
```

The `db` folder contains about 4000 files and has a size of ~230MB

Version info:
  - MBT Version: 1.0.16
  - OS Version: Windows 10 version 1909

Please let me know if further information is required. Thanks for your help!

Best Regards,
Florian

**Comment by d021248** [comment]
We are observing the same issue. The `mtb build` command is getting stuck when trying to build the following module:
```
- name: mkv-api-broker
  type: nodejs
  path: broker
```

The build starts, but never returns after the last output `[2020-11-15 16:46:54]  INFO ...` below:
```
[2020-11-15 16:46:51]  INFO validating the MTA project
[2020-11-15 16:46:51]  INFO building the "mkv-api-broker" module...
[2020-11-15 16:46:51]  INFO executing the "npm install --production" command...

up to date in 0.786s

[2020-11-15 16:46:54]  INFO the build results of the "mkv-api-broker" module will be packaged and saved in the "C:\SAPDevelop\c4u-edom-mkv\.c4u-edom-mkv_mta_build_tmp\mkv-api-broker" folder
```

We have noticed, that `mtb version 1.0.15` successfully completes the job, while `mbt version 1.0.16` causes the problem.

**Comment by d021248** [comment]
Steps to reproduce the issue:
* clone mkv github [repo](https://github.wdf.sap.corp/c4u/c4u-edom-mkv)
* execute `mbt build -t .`
* the affected module is `mkv-api-broker` (move the module at the front of the `mta.yaml` to avoid waiting)

The problem occurs almost every time on my windows pc.
Other colleagues observe it sporadically.

**Comment by jofunkillo** [comment]
I have the same problem building an MTA application. When generating final zip packages MBT tool just stop working, no log information about the problem (zip file is 0 bytes). Before, it just happened to me 1 of each 10 times, now it happen every time and I can't build my project.

**Comment by michawai** [comment]
@jofunkillo can you provide your project?

**Comment by jofunkillo** [comment]
Sorry, I can't share for privacy reasons. 

As mentioned above, I downloaded the MBT version 1.0.15 and the whole project compiled correctly now.

My project is having 7 UI applications, 4 fiori freestyle applications and 3 fiori elements apps, on top of a CAP service. It takes a while to build it.

With latest MBT version, the build process was randomly stopping in the generation of the zip files. I tried to delete the existing files before the build process to see if that could be the reason (file locks) but sometimes it didn't work.

**Comment by npetralia** [comment]
Hi,

Got exactly the same issue than @jofunkillo 
When generating final zip packages MBT tool just stop working, no log information about the problem but zip is not 0 size. It looks like an issue with McAfee as my CPU goes to 100%.
Everything was stuck but mbt was running because when i want to remove the zip file it says that mbt.exe lock the file.

Goes back to version 1.0.15 solve the issue.

make -v
GNU Make 4.3
Built for Windows32
Copyright (C) 1988-2020 Free Software Foundation, Inc.
License GPLv3+: GNU GPL version 3 or later <http://gnu.org/licenses/gpl.html>
This is free software: you are free to change and redistribute it.
There is NO WARRANTY, to the extent permitted by law.

Cloud MTA Build Tool version 1.0.15

|:------------------ | ----------- |
| Node.js            | v12.19.0    |
| @sap/cds           | 4.4.5       |
| @sap/cds-runtime   | 2.7.5       |
| @sap/cds-compiler  | 1.46.6      |
| @sap/cds-dk        | 3.3.2       |

Thanks

**Comment by d019820** [comment]
Hello colleagues,

I have the same issue, i.e. I can't build my project **https://github.wdf.sap.corp/GRC-DPP/questionnaire** anymore with mbt 1.016

Process stucks when creating data.zip file. Sometimes the first file can be created and process stucks on the second file, as can be seen from the screen shot.  MBT tool just stops working w/o any log information about the problem.

Thanks, Thomas

![image](https://user-images.githubusercontent.com/20061381/101766444-189f1c80-3ae3-11eb-84d4-389bee4047a9.png)

**Comment by michawai** [comment]
Issue was reproduced and we're investigating.

**Comment by d019820** [comment]
Hello colleagues,
just did an update to mbt version 1.0.17 and ran the build process multiple times. It came to a successfull end every time.
Thanks, Thomas

**Comment by michawai** [comment]
Issue solved by version 1.0.17.
